server/src/main/java/org/apache/iotdb/db/qp/physical/sys/LoadConfigurationPlan.java

```diff
@@ -17,19 +17,107 @@
 
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Properties;
+import org.apache.iotdb.db.exception.query.QueryProcessException;
 import org.apache.iotdb.db.qp.logical.Operator.OperatorType;
 import org.apache.iotdb.db.qp.physical.PhysicalPlan;
 import org.apache.iotdb.tsfile.read.common.Path;
-
-import java.util.List;
+import org.apache.iotdb.tsfile.utils.ReadWriteIOUtils;
 
 public class LoadConfigurationPlan extends PhysicalPlan {
 
-  public LoadConfigurationPlan() {
-    super(false);
-    this.setOperatorType(OperatorType.LOAD_CONFIGURATION);
+  // an array of properties, the size of which is always 2.
+  // The first element is the properties for iotdb-engine, the second
+  // is for cluster-config
+  private Properties[] propertiesArray;
+
+  LoadConfigurationPlanType loadConfigurationPlanType;
+
+  public LoadConfigurationPlan(LoadConfigurationPlanType loadConfigurationPlanType,
+      Properties[] propertiesArray)
+      throws QueryProcessException {
+    super(false, OperatorType.LOAD_CONFIGURATION);
+    if (loadConfigurationPlanType != LoadConfigurationPlanType.GLOBAL) {
+      throw new QueryProcessException(
+          "The constructor with 2 parameters is for load global configuration");
+    }
+    if (propertiesArray.length != 2) {
+      throw new QueryProcessException("The size of propertiesArray is not 2.");
+    }
+    this.loadConfigurationPlanType = loadConfigurationPlanType;
+    this.propertiesArray = propertiesArray;
+  }
+
+  public LoadConfigurationPlan(LoadConfigurationPlanType loadConfigurationPlanType)
+      throws QueryProcessException {
+    super(false, OperatorType.LOAD_CONFIGURATION);
+    if (loadConfigurationPlanType != LoadConfigurationPlanType.LOCAL) {
+      throw new QueryProcessException(
+          "The constructor with 1 parameters is for load local configuration");
+    }
+    this.loadConfigurationPlanType = loadConfigurationPlanType;
+  }
+
+  // only for deserialize
+  public LoadConfigurationPlan(){
+    super(false, OperatorType.LOAD_CONFIGURATION);
+  }
+
+  @Override
+  public void serialize(DataOutputStream stream) throws IOException {
+    int type = PhysicalPlanType.LOAD_CONFIGURATION.ordinal();
+    stream.writeByte((byte) type);
+    stream.writeInt(loadConfigurationPlanType.ordinal());
+    if (loadConfigurationPlanType == LoadConfigurationPlanType.GLOBAL) {
+      stream.writeInt(propertiesArray.length);
+      for (Properties properties : propertiesArray) {
+        if (properties == null) {
+          stream.writeInt(0);
+        } else {
+          stream.writeInt(1);
+          Map<String, String> propertiesMap = new HashMap<>();
+          for (Entry<Object, Object> entry : properties.entrySet()) {
+            propertiesMap.put(String.valueOf(entry.getKey()), String.valueOf(entry.getValue()));
+          }
```

Comment:
I wonder if it is necessary to put the properties into a new map, as you have the entrySet, you can just write them into the stream within this loop.

Reply:
Thanks. It is not necessary.